content/blog/2016/2016-05-19-update-plugin-for-pipeline-tutorial.md

```diff
@@ -0,0 +1,189 @@
+
+ ![archive metastep snippet](images/2016-05-19-update-plugin-for-pipeline-tutorial/15_archive_metastep_snippet.png)
+
+From the snippet generator we can see that it's possible to trigger an Archive action with syntax like:
+
+    step([$class: 'ArtifactArchiver', artifacts: 'foo*', excludes: null])
+
+This is the "metastep".  It's a way to trigger any build action that implements `SimpleBuildStep`, without having to actually implement a for-realz "step" that extends the Pipeline DSL like we did above.  In many cases, it might only make sense to do one or the other in your plugin; you probably don't really need both.
+
+For the purposes of this tutorial, we're going to do both.  For a couple of reasons:
+
+1. Why the heck not?  :)  It's a good demonstration of how the metastep stuff works.
+2. Because implementing the "for realz" step will be a lot easier if the Gatling action that we're trying to call from our `gatlingArchive()` syntax is using the newer Jenkins APIs that are required for subclasses of `SimpleBuildStep`.
+
+`GatlingPublisher` is the main build action that we're interested in using in Pipeline jobs.  So, with all of that in mind, here's our next goal: get `step([$class: 'GatlingPublisher', ...)` showing up in the Snippet Generator.
+
+The [javadocs for the SimpleBuildStep class](https://github.com/jenkinsci/jenkins/blob/jenkins-1.651.2/core/src/main/java/jenkins/tasks/SimpleBuildStep.java#L51-L66) have some notes on what you need to do when porting an existing `Builder` or `Publisher` over to implement the `SimpleBuildStep` interface.  In all likelihood, most of what you're going to end up doing is to replace occurrences of `AbstractBuild` with references to the `Run` class, and replace occurrences of `AbstractProject` with references to the `Job` class.  The APIs are pretty similar, so it's not too hard to do once you understand that that's the game.  There is some discussion of this in the [pipeline-plugin DEVGUIDE](https://github.com/jenkinsci/pipeline-plugin/blob/893e3484a25289c59567c6724f7ce19e3d23c6ee/DEVGUIDE.md#historical-background).
+
+For the Gatling plugin, my [initial efforts to port the `GatlingPublisher` over to implement `SimpleBuildStep`](https://github.com/cprice404/gatling-plugin/commit/288041c696840ea8eaf21705caf756d3d4bb1f94) only required the `AbstractBuild` -> `Run` refactor.
+
+After making these changes, I fired up the development Jenkins server, and, voila!
+
+![gatling metastep snippet](images/2016-05-19-update-plugin-for-pipeline-tutorial/20_gatling_metastep_snippet.png)
+
+So, now, we can add a line like this to a Pipeline build script:
+
+    step([$class: 'GatlingPublisher', enabled: true])
```

Comment:
That's a fair point; seems like for the sake of the blog post it's maybe useful to have a non-zero number of params though :)